src/frontend/main_window.rs

```diff
@@ -184,12 +268,16 @@ fn get_total_view_size(monitors: &Vec<&gdk4::Monitor>) -> (i32, i32) {
     (width, height)
 }
 
-fn register_keyboard_events(window: &gtk::Window) {
+fn register_keyboard_events(window: &gtk::Window, mode: cli::Command) {
     let event_controller = gtk::EventControllerKey::new();
+    let window2 = window.clone();
 
-    event_controller.connect_key_pressed(|_, key, _, _| {
+    event_controller.connect_key_pressed(move |_, key, _, _| {
         if let gdk4::Key::Escape = key {
-            std::process::exit(0);
+            match mode {
+                cli::Command::Gui => std::process::exit(0),
```

Comment:
This won't do a clean shutdown. iirc there is something similar via `relm4::main_application`